### PR TITLE
Added checksum to summary to make it easy to verify two runs will run…

### PR DESCRIPTION
**Pull request by anderskvist**
… on the same zonefile.

### CODE REVIEW

**Comment by abrander** [comment]
This is interesting. What is the use case for this?

Is it compatible with `sha256sum`?

**Comment by anderskvist** [comment]
@abrander the case was that I had to get the changes approved before applying them and a LOT of them. So instead of me having to verify every single line in a few days when I expected to get the "go", I would instead just be able to check the checksum and go on :)

```
$ sha256sum zone
5ec8754e939ff941f0c21173e1adcb8db31778e1844ecb2fd315a00526dfcfb5  zone
$ ~/go/bin/cfzone zone
Records to delete:
...

Records to add:
...

Summary:
Zone checksum: 5ec8754e939ff941f0c21173e1adcb8db31778e1844ecb2fd315a00526dfcfb5
Records to delete: 1
Records to add: 1
Records to update: 0
Unchanged records: XXXX
2 change(s). Continue (y/N)? ^C
$ 
```

It is compatible with sha256sum, yes :)

File: main.go [hunk 1]
```diff
@@ -66,6 +69,15 @@ func main() {
 		exit(1)
 	}
 
+	hasher := sha256.New()
+	s, err := ioutil.ReadFile(path)
```
Comment (abrander) [inline main.go]: I think it's not so nice to open the same file multiple times. How about something like this:
```go
hasher := sha256.New()
_, err = io.Copy(hasher, f)
if err != nil {
	fmt.Fprintf(stderr, "Error reading '%s': %s\n", "path", err.Error())
	exit(1)
}
f.Seek(0, 0)
```

File: main.go [hunk 2]
```diff
@@ -66,6 +69,15 @@ func main() {
...
+	hasher.Write(s)
+	if err != nil {
+		fmt.Fprintf(stderr, "Error calculating checksum for '%s': %s\n", path, err.Error())
+		exit(1)
+	}
+	zoneChecksum := hex.EncodeToString(hasher.Sum(nil))
```
Comment (abrander) [inline main.go]: `hex.EncodeToString` is not needed. `fmt.Printf` can format properly using `%x`.

File: main.go [hunk 3]
```diff
@@ -133,6 +145,7 @@ func main() {
 		}
 
 		fmt.Fprintf(stdout, "Summary:\n")
+		fmt.Fprintf(stdout, "Zone checksum: %s\n", zoneChecksum)
```
Comment (abrander) [inline main.go]: How about "SHA256 zone checksum" to make it explicit what the value represent?

**Review by abrander — CHANGES_REQUESTED** [review]
Just a few suggestions.

File: main.go (hunk 3 above)
Comment (anderskvist) [reply main.go]: Added

File: main.go (hunk 2 above)
Comment (anderskvist) [reply main.go]: done!

File: main.go (hunk 1 above)
Comment (anderskvist) [reply main.go]: I agree, it's been fixed

**Review by abrander — APPROVED** [review]
LGTM! :-)

**Pull request merged**